<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <ros2_control name="RealRobot" type="system">
        <hardware>
            <plugin>mock_components/GenericSystem</plugin>
        </hardware>
        
        <joint name="front_right_wheel_steer_joint">
            <command_interface name="position"/>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
        </joint>
        
        <joint name="front_left_wheel_steer_joint">
            <command_interface name="position"/>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
        </joint>

        <joint name="front_right_wheel_joint">
            <command_interface name="position"/>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
        </joint>
        
        <joint name="front_left_wheel_joint">
            <command_interface name="position"/>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
        </joint>

        <joint name="back_right_wheel_joint">
            <command_interface name="velocity">
                <param name="min">-22</param>
                <param name="max">22</param>
            </command_interface>
            <state_interface name="velocity">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
        </joint>
        
        <joint name="back_left_wheel_joint">
            <command_interface name="velocity">
                <param name="min">-22</param>
                <param name="max">22</param>
            </command_interface>
            <state_interface name="velocity">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
        </joint>
    </ros2_control>

</robot>
